### ISSUE TITLE
Address Nested ConstraintLayouts in search & tabs tray experiences

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/fenix#15788]
~~Blocked by #15280 & #15644.~~ They are addressed.

20ms tabs tray improvement.

Please remove the nested `ConstraintLayout`s in the search screen and tabs tray. From a correctness and performance perspective, ConstraintLayouts are designed to minimize nesting which improves perf and reduces battery use. Ideally, ConstraintLayout is the root element and contains only leaf nodes.

We added a custom lint rule https://github.com/mozilla-mobile/fenix/issues/15280 to check for nested ConstraintLayouts. In order to work on this bug:
- Delete the `app/lint-baseline.xml` file (it only contains these violations)
- Run `./gradlew lintDebug` and check the violation output
- Fix the associated errors; be careful about how accessibility may change
- (when completed) Create a new baseline file with any remaining violations

**WARNING: replacing a ConstraintLayout with several nested ViewGroups is unlikely to improve performance. Remember, a goal is to reduce nesting. If you're unable to remove the extra ConstraintLayout(s) without doing this, it's probably better to suppress this violation.**

Please ask the performance team if you have questions!

**Comment by liuche** [comment]
Two screens involved. Can also run some performance tests (ask the perf team) afterwards.

**Comment by mcarare** [comment]
For QA: There is no specific testing is needed, just check that the layout changes did not introduce any regressions in behavior. Thank you!

**Comment by sflorean** [comment]
Tested on latest Nightly 11/19 and we didn't encounter any new issue. Devices:
- Pixel 3 (Android 11)
- LG G7 (Android 8.0).
